gui/src/ScenarioSelector.tsx

```diff
@@ -9,31 +65,332 @@ interface ScenarioSelectorProps {
   defaultShowPrimaryFlag: boolean;
   showPrimaryFlag?: boolean;
   scenarios?: LoV;
+  defaultScenarios?: LoV;
   defaultScenarioId?: string;
   scenarioId?: string;
   onScenarioCreate?: string;
   coreChanged?: Record<string, unknown>;
   updateVarNames: string;
 }
 
+const scenarioSchema = Yup.object().shape({
+  config: Yup.string()
```

Comment:
Interresting.... using Yup to validate schemas.
I'm not sure that is useful since we're on the lib side.
But please comment.